### ISSUE TITLE
Additional information on kptncook API

### ISSUE THREAD

**Issue by gloriousDan** [ephes/kptncook#11]
Nice project :)
I was also inspired by the blog post by fhantke and did some reversing of the app myself.
I posted the findings here: https://github.com/gloriousDan/kptncook-api-reverse-engineering

You may find it interesting since a couple more endpoints are included in my writedown :)

There's also https://github.com/hhursev/recipe-scrapers which supports scraping kptncook recipes from their share link since version v13.19.0.

Feel free to close this issue :)

**Comment by gloriousDan** [comment]
I also tried analyzing with only the apk but, just as it was for you, I didn't find much. In the end I set up a android virtual device, installed kpntcook on it and routed all traffic through a [burp](https://portswigger.net/burp) proxy. To intercept https traffic I additionally had to install the burp CA on the phone. After doing all that I was able to see the plain http calls.

I'm looking forward to the update, maybe a function which exports all favorite recipes of a user would be great too. That would be great for backing the kptncook recipes up. 

No rush though enjoy the pycon :)

**Comment by ephes** [comment]
Integrated most of the new api endpoints. You should be able now to backup your favorites, search for recipes by id etc. I didn't adapt the new receipt format with lang set, because I have to modify the internal recipe representation then. Maybe I'll do it later on. Why do you think this is the new format? And do you know where I can find the endpoint for the daily recipes in the new format?

If you have any recommendations about how to name cli arguments/options or how it should behave, they would be very welcome :).

**Comment by gloriousDan** [comment]
Great, this looks amazing. I'll definetly try this out later to archive my kptncook collection. 

I first started reversing the API with an old version of the app. Later I switched to a newer apk version and then discovered the new format. Unfortunately I'm not sure which versions I used and when the API was changed.

Concerning the daily endpoint in the new format. I didn't really have time to look yet but it seems like appending the `lang` parameter does the trick. I didn't check if it's the same format as the new search endpoint's format yet though.

Maybe everything concerning the ACCESS_TOKEN can be renamed into Personal access token so nobody confuses it with the kptnkey. A user doesn't really need to concern himself with the kptnkey though so it's probably fine. Other than the `http` option I think everything is named pretty clearly.

Thank you very much for your work on this :)
